tests/unit/EntityId/PrefixingEntityIdParserTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\BasicEntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixingEntityIdParser;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\SuffixEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefixingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @dataProvider validInputProvider
+	 */
+	public function testParse( $prefix, $input, $expected ) {
+		$parser = new PrefixingEntityIdParser( $prefix, new BasicEntityIdParser() );
+		$this->assertEquals( $expected, $parser->parse( $input ) );
+	}
+
+	public function validInputProvider() {
+		return array(
+			array( 'wikidata:', 'Q1337', new ItemId( 'wikidata:Q1337' ) ),
+			array( 'foo:', 'P123', new PropertyId( 'foo:P123' ) ),
+			array( 'foo:', 'bar:Q1337', new ItemId( 'foo:bar:Q1337' ) ),
+		);
+	}
+
+	public function testParse_prefixIncluded() {
+		$parser = new PrefixingEntityIdParser( 'wikidata:', new BasicEntityIdParser() );
+		$this->assertEquals( new ItemId( 'wikidata:Q1337' ), $parser->parse( 'wikidata:Q1337' ) );
+	}
+
+	/**
+	 * @dataProvider invalidInputProvider
+	 */
+	public function testParse_invalidInput( $prefix, $input ) {
+		$parser = new PrefixingEntityIdParser( $prefix, new BasicEntityIdParser() );
+		$this->setExpectedException( EntityIdParsingException::class );
```

Comment:
Travis is still running the tests with 5.3 and 5.4. Time to bump to 5.5+... either as a separate PR or in here, whichever is easier